tests/components/buienradar/test_weather.py

```diff
@@ -28,3 +31,148 @@ async def test_smoke_test_setup_component(
 
     state = hass.states.get("weather.buienradar")
     assert state.state == "unknown"
+
+
+async def test_smoke_test_setup_component_no_coordinates(
+    aioclient_mock: AiohttpClientMocker, hass: HomeAssistant
+) -> None:
+    """Smoke test for successfully set-up with default config."""
+    aioclient_mock.get(
+        "https://data.buienradar.nl/2.0/feed/json", status=HTTPStatus.NOT_FOUND
+    )
+
+    mock_entry = MockConfigEntry(
+        domain=DOMAIN, unique_id="TEST_ID", data=TEST_CFG_INVALID_COORDINATES
+    )
+
+    mock_entry.add_to_hass(hass)
+
+    await hass.config_entries.async_setup(mock_entry.entry_id)
+    await hass.async_block_till_done()
+
+    state = hass.states.get("weather.buienradar")
+    assert state is None
+
+
+async def test_smoke_test_data(
+    aioclient_mock: AiohttpClientMocker, hass: HomeAssistant
+) -> None:
+    """Smoke test for successfully set-up with default config and check fetcheddata."""
+    brJson = load_json_object_fixture("buienradar.json", DOMAIN)
+    raindata = load_fixture("raindata.txt", DOMAIN)
+    aioclient_mock.get(
+        "https://data.buienradar.nl/2.0/feed/json", status=HTTPStatus.OK, json=brJson
+    )
+    aioclient_mock.get(
+        f"https://gpsgadget.buienradar.nl/data/raintext?lat={TEST_CFG_DATA[CONF_LATITUDE]:.2f}&lon={TEST_CFG_DATA[CONF_LONGITUDE]:.1f}",
+        status=HTTPStatus.OK,
+        text=raindata,
+    )
+
+    mock_entry = MockConfigEntry(domain=DOMAIN, unique_id="TEST_ID", data=TEST_CFG_DATA)
+
+    mock_entry.add_to_hass(hass)
+
+    await hass.config_entries.async_setup(mock_entry.entry_id)
+    await hass.async_block_till_done()
+
+    state = hass.states.get("weather.buienradar")
+    assert state.state == "cloudy"
+
+    assert state.attributes["temperature"] == 20.4
+    assert state.attributes["apparent_temperature"] == 20.5
+    assert state.attributes["temperature_unit"] == "°C"
+    assert state.attributes["humidity"] == 76.0
+    assert state.attributes["pressure"] == 1020.0
+    assert state.attributes["pressure_unit"] == "hPa"
+    assert state.attributes["wind_bearing"] == 38
+    assert state.attributes["wind_gust_speed"] == 23.4
+    assert state.attributes["wind_speed"] == 14.4
+    assert state.attributes["wind_speed_unit"] == "km/h"
+    assert state.attributes["visibility"] == 32.6
+    assert state.attributes["visibility_unit"] == "km"
+    assert state.attributes["precipitation_unit"] == "mm"
+
```

Comment:
Would be good to include an action call to get forecast data as well

Reply:
added testcase calling the service and snapshotting the data returned